addons/account_peppol/models/res_config_settings.py

```diff
@@ -240,46 +240,29 @@ def button_cancel_peppol_registration(self):
         Sets the peppol registration to canceled
         - If the user is active on the SMP, we can't just cancel it.
           They have to request a migration key using the `button_migrate_peppol_registration` action
+          or deregister.
         - 'not_registered', 'rejected', 'canceled' proxy states mean that canceling the registration
           makes no sense, so we don't do it
         - Calls the IAP server first before setting the state as canceled on the client side,
           in case they've been activated on the IAP side in the meantime
         """
         self.ensure_one()
+        # check if the participant has been already registered
+        self.env.ref('account_peppol.ir_cron_peppol_get_participant_status')._trigger()
+        if not tools.config['test_enable'] and not modules.module.current_test:
+            self.env.cr.commit()
 
         if self.account_peppol_proxy_state == 'active':
-            raise UserError(_("Can't cancel an active registration. Please request a migration instead."))
+            raise UserError(_("Can't cancel an active registration. Please request a migration or deregister instead."))
 
         if self.account_peppol_proxy_state in {'not_registered', 'rejected', 'canceled'}:
             raise UserError(_(
                 "Can't cancel registration with this status: %s", self.account_peppol_proxy_state
             ))
 
         self._call_peppol_proxy(endpoint='/api/peppol/1/cancel_peppol_registration')
-        self.account_peppol_proxy_state = 'canceled'
-
-    def button_reopen_peppol_registration(self):
```

Comment:
You should not remove a method in stable version. Instead, add a `deprecated` comment and remove it during the fw-port in master